src/sexp.c

```diff
@@ -50,3 +62,27 @@ bool is_named(SEXP x) {
 
   return true;
 }
+
+
+SEXP r_missing_arg() {
+  return R_MissingArg;
+}
+bool r_is_missing(SEXP x) {
+  return x == R_MissingArg;
+}
+
+
+SEXP rlang_is_null(SEXP x) {
+  return r_scalar_lgl(r_is_null(x));
+}
+bool r_is_null(SEXP x) {
+  return x == R_NilValue;
+}
+
+
+SEXP r_duplicate(SEXP x) {
+  return Rf_duplicate(x);
+}
+SEXP r_shallow_duplicate(SEXP x) {
```

Comment:
Should this be an argument to `r_duplicate()`? That would force you to consider whether it should be shallow or deep whenever you duplicate.